src/server/pkg/worker/api_server.go

```diff
@@ -490,6 +489,22 @@ func HashDatum(pipelineInfo *pps.PipelineInfo, data []*Input) (string, error) {
 		hash.Write(datum.FileInfo.Hash)
 	}
 
+	hash.Write([]byte(pipelineInfo.Pipeline.Name))
+	hash.Write([]byte(pipelineInfo.Salt))
+
+	return client.HashPipelineName(pipelineInfo.Pipeline.Name) + hex.EncodeToString(hash.Sum(nil)), nil
```

Comment:
What's the point of including pipeline name in the hash if we are including the salt?  It's a UUID right?